### ISSUE TITLE
ls doesn't work properly inside of a block

### ISSUE THREAD

**Issue by Himal-Adhikari** [nushell/nushell#8929]
### Describe the bug

While trying to change the LS_COLORS environment variable to different values and executing ls to see the corresponding output, if an echo command was executed after ls, ls would not get executed, but if the echo command was before ls or omitted entirely, ls would be executed

### How to reproduce

1. Download [vivid](https://github.com/sharkdp/vivid)
2. execute this code `for theme in (vivid themes | lines) {
print ("Theme: " + $theme)
let-env LS_COLORS = (vivid generate $theme)
ls
echo
}`

### Expected behavior

ls should be executed in all three versions

### Screenshots

With echo at the end
![initial](https://user-images.githubusercontent.com/120376987/233008509-fa411230-1035-4627-a8a6-e3560e9c6f0d.png)

With echo before ls
![echo on top](https://user-images.githubusercontent.com/120376987/233008648-4fc3b6ac-9a5d-4f7d-94a0-c76375fb7adb.png)

Without echo
![without echo](https://user-images.githubusercontent.com/120376987/233008777-f174f741-5830-4944-bfb5-9c18a352fdc7.png)

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.78.0                                   |
| branch             |                                          |
| commit_hash        | 2ec20286371f1d90a77fdd5d82818c6c5851f090 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)      |
| rust_channel       | 1.66.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)      |
| build_time         | 2023-04-04 19:13:32 +00:00               |
| build_rust_channel | release                                  |
| features           | default, zip                             |
| installed_plugins  |                                          |

**Comment by WindSoilder** [comment]
Yeah it's expected, actually it should only print out `Theme: xxx`.  You can refer to https://github.com/nushell/nushell/pull/8618 for more details

For printing inside for loop, you need to use `print (ls)`

**Comment by kubouch** [comment]
Broken LS_COLORS is a known issue due to pipeline metadata getting lost when converted from pipeline to value.

**Comment by fdncred** [comment]
But this will work for seeing what the themes look like. I use vivid too sometimes.
```
vivid themes | lines | each {|theme|  
  print $"Theme ($theme)"
  let-env LS_COLORS = (vivid generate $theme)
  print (ls | table)
  }
```

previous issues
https://github.com/nushell/nushell/issues/8030
https://github.com/nushell/nushell/issues/7169
https://github.com/nushell/nushell/issues/4501
https://github.com/nushell/nushell/issues/4319
